app/lib/core/models/user_info.dart

```diff
@@ -0,0 +1,34 @@
+import 'package:flutter/material.dart';
+
+class UserInformation {
+  final String userId;
+  final String firstName;
+  final String lastName;
+  final String userEmailAddress;
+  final String userPhoneNumber;
+
+  UserInformation({
+    @required this.userId,
+    @required this.firstName,
+    @required this.lastName,
+    @required this.userEmailAddress,
+    @required this.userPhoneNumber,
+  });
+
+  UserInformation.fromFirestore(Map<String, dynamic> data)
```

Comment:
We check if data is null in `HistoryRecord`. Should we not also be doing that here?

Reply:
This applies to services already and therefore not need to be in the model itself. We check if snapshot has data, if it's not null, then we call fromFirestore

Reply:
I'm sorry, could you explain further? I'm not sure what "this applies to services already" means.

Reply:
I'm referring to services like authentication when it's checking if userinfo/ Firebase user is null or not.

Reply:
I understand what you're saying as: "we're already checking if it's null in another class". I think we should be consistent with null checks: either we check if null before making the class, or do it in the constructor. 

- [ ] So, we should probably not accept null data in `HistoryRecord fromFirestore()` and check that the data is not null before using it.